### PR TITLE
Introduce support for Ruby

### PR DESCRIPTION
**Pull request by fallwith**
Add content for building and publishing layers for Ruby based Lambda functions. See `ruby/README.md` for more detailed information.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,19 +26,25 @@ cd ..
 ```
 
 ```
-cd nodejs;
```
Comment (mrickard) [inline README.md:29]: Wow--not sure how those got inserted, but thank you for removing them!

Comment (fallwith) [reply README.md:29]: Sure thing! Perhaps they were leftover from having all of the commands on one line and the semicolons were needed as delimiters. They weren't hurting anything, but I figured I'd remove them to have all of the language examples aligned with not having them.

File: ruby/.gitignore [hunk 2]
```diff
@@ -0,0 +1,6 @@
+.bundle/
+dist/
+log/
+support/node_modules/
+support/log/
+Gemfile.lock
```
[the remark this answers is not in the document]
Comment (fallwith) [reply ruby/.gitignore:6]: > Do any of your team members use IntelliJ or another JetBrains editor? You might want to add `.idea` to your ignore.

Good call, but no we don't use `.idea` based editors or any software shared across the whole team that would need to be represented here.

**Comment by fallwith** [comment]
I'm ready to share a layer with a customer for beta testing! Thanks again, @mrickard!

Screenshot 2024-03-01 at 7 19 22 PM

**Comment by mrickard** [comment]
We'll want to make sure the automation is wired up for this. 

For running automated tests on push/PR, we currently have GitHub Actions at https://github.com/newrelic/newrelic-lambda-layers/tree/master/.github/workflows ...though this could be made a little more responsive. For example, right now you're getting a lot of unreported status messages, because the changes aren't in the Node repo, and Node tests are skipped. (As they should be for that scenario...but here it presents as incomplete tests.)

Currently the layer publishing jobs are defined in https://github.com/newrelic/newrelic-lambda-layers/tree/master/.circleci/config.yml ...which I'm currently porting to GHA.

The build jobs may need a Dockerfile for Ruby. (If not, great! But Java and Node currently use them.) If so, those are defined in https://github.com/newrelic/newrelic-lambda-layers/tree/master/dockerfiles

I don't expect you'd need to change the makefile, but you may want to take a look there.

**Comment by fallwith** [comment]
> The build jobs may need a Dockerfile for Ruby. (If not, great! But Java and Node currently use them.) If so, those are defined in https://github.com/newrelic/newrelic-lambda-layers/tree/master/dockerfiles

I'm curious about the approach Python takes, as I don't see Dockerfiles for the supported Python runtimes.

For building the Ruby layers, I have made the build script fairly fault tolerant to a Ruby version that doesn't match the supported AWS Ruby Lambda runtime(s) (only Ruby v3.2 at this time), but it would be great to build things using the same Ruby version that we are targetting. If Docker helps with that, I welcome it.

**Comment by fallwith** [comment]
> For building the Ruby layers, I have made the build script fairly fault tolerant to a Ruby version that doesn't match the supported AWS Ruby Lambda runtime(s) (only Ruby v3.2 at this time), but it would be great to build things using the same Ruby version that we are targetting. If Docker helps with that, I welcome it.

Ok, @mrickard. I've added a Ruby Dockerfile (only one because only one Lambda runtime exists) with 
be426c6.

**Comment by mrickard** [comment]
@fallwith This is looking good to me! This coming week I should be able to check in a fork, and we can verify built-layer behavior.

**Pull request merged**